### ISSUE TITLE
Possible query performance regression introduced in 0.20?

### ISSUE THREAD

**Issue by connorlay** [tree-sitter/tree-sitter#1510]
Neovim nightly recently upgraded from tree-sitter `0.19` to `0.20`, which introduced a significant syntax highlighting performance regression for Elixir. After a bit of debugging, I think I located the bottleneck: a call to [`ts_query__analyze_patterns`](https://github.com/tree-sitter/tree-sitter/blob/v0.20.1/lib/src/query.c#L772), executed when [neovim loads the query file](https://github.com/neovim/neovim/blob/a68faed02dc8e37b8f10da14dc02e33e6ed93947/src/nvim/lua/treesitter.c#L1183). From what I can tell, it seems neovim is calling `ts_query_new` each time a new Elixir buffer is created, resulting in users hitting this bottleneck frequently when navigating an Elixir project. You can find more details in the related [nvim-treesitter issue](https://github.com/nvim-treesitter/nvim-treesitter/issues/1957#issuecomment-980511149), including reproduction steps for neovim nightly and [tree-sitter-elixir](https://github.com/elixir-lang/tree-sitter-elixir).

I'm interested in solving this problem, but I'm not sure where to start. Should this be considered a performance regression in tree-sitter `0.20` and addressed here? Or should neovim implement some kind of caching mechanism to avoid multiple calls to `ts_query_new` now that the behavior has changed? I suspect https://github.com/tree-sitter/tree-sitter/commit/26dac9b2dd52c9c3998f600cad85dc7d4b63974a introduced this issue as that was the most recent change to `ts_query__analyze_patterns`.

Thanks for the help!

cc @theHamsta @jonatanklosko

**Comment by alexander-p30** [comment]
I don´t know if it helps or not, but: it seems the problem only occurs when compiling neovim with tree-sitter ```0.20.1```, as building it with tree-sitter ```0.20.0``` solves the performance regression for me.

**Comment by the-mikedavis** [comment]
I've been noticing this in [helix](https://github.com/helix-editor/helix) as well: initially loading the queries takes much longer (5-10 secs vs ~1 sec on a debug build), but the speed of tree-sitter otherwise seems unaffected.

I ran a bisect and it looks like the regression came from 1fe0420f0fe3df633c4f8b80d0c991a7aa214eeb. Reverting that commit but otherwise keeping changes since v0.20.0 seems to be as fast as ever

**Comment by theHamsta** [comment]
@clason would it be worth to continue using 0.20.0 for 0.6 nvim while we're figuring what's going on? The update to 0.20.0 was pretty close to the release

**Comment by maxbrunsfeld** [comment]
I think this may be fixed by https://github.com/tree-sitter/tree-sitter/commit/25f64e1eb66bb1ab3eccd4f0b7da543005f3ba79. When  you get a chance, could you check to see if that fixes the problem?

**Comment by the-mikedavis** [comment]
v0.20.0: 1.61s
v0.20.1: 19.93s
25f64e1: 3.39s

I just gave this a try with debug builds of helix (`cargo build`) and I see the time to start up on an elixir file go:

- v0.20.0: 1.61s
- v0.20.1: 19.93s
- 25f64e1: 3.39s

Once 25f64e1 is compiled into a release build though (`cargo build --release`), it's virtually instant and indistinguishable from v0.20.0 :tada: 

So I'd say this fix works for me, but I'd await the neovim folks' results to be sure :+1:

**Comment by clason** [comment]
> So I'd say this fix works for me, but I'd await the neovim folks' results to be sure 👍

Can you give the linked PR a try?

**Comment by alexander-p30** [comment]
> > So I'd say this fix works for me, but I'd await the neovim folks' results to be sure +1
> 
> Can you give the linked PR a try?

It's way better, but not as fast as it used to be. I've compiled neovim in the ```bump-treesitter``` branch and ran nvim --startuptime for that build and for neovim 0.5.1, here's the output of both:

**0.7 with tree-sitter@25f64e1**
```
times in msec
 clock   self+sourced   self:  sourced script
 clock   elapsed:              other lines

000.008  000.008: --- NVIM STARTING ---
000.294  000.286: locale set
000.625  000.331: inits 1
000.642  000.017: window checked
004.170  003.528: parsing arguments
004.252  000.082: expanding arguments
004.273  000.021: inits 2
004.799  000.526: init highlight
004.801  000.002: waiting for UI
005.737  000.935: done waiting for UI
005.766  000.030: init screen for UI
005.787  000.021: init default mappings
005.863  000.076: init default autocommands
008.119  001.701  001.701: sourcing /home/alexander/.local/share/nvim/site/autoload/plug.vim
018.314  000.063  000.063: sourcing /home/alexander/.vim/plugged/vim-elixir/ftdetect/elixir.vim
018.579  000.033  000.033: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/elixir.vim
018.613  000.014  000.014: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fennel.vim
018.653  000.021  000.021: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fish.vim
018.686  000.017  000.017: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fusion.vim
018.723  000.020  000.020: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gdresource.vim
018.751  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gdscript.vim
018.780  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/glimmer.vim
018.833  000.036  000.036: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/glsl.vim
018.867  000.017  000.017: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gomod.vim
018.907  000.023  000.023: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/graphql.vim
018.947  000.023  000.023: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/hcl.vim
018.989  000.023  000.023: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/heex.vim
019.021  000.016  000.016: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/hjson.vim
019.057  000.016  000.016: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/json5.vim
019.092  000.015  000.015: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/julia.vim
019.136  000.025  000.025: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/ledger.vim
019.171  000.017  000.017: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/nix.vim
019.207  000.015  000.015: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/prisma.vim
019.246  000.017  000.017: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/pug.vim
019.283  000.017  000.017: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/ql.vim
019.340  000.037  000.037: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/query.vim
019.383  000.022  000.022: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/surface.vim
019.416  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/teal.vim
019.469  000.033  000.033: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/tlaplus.vim
019.501  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/yang.vim
019.533  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/zig.vim
019.834  009.922  009.340: sourcing /usr/local/share/nvim/runtime/filetype.vim
019.972  000.057  000.057: sourcing /usr/local/share/nvim/runtime/ftplugin.vim
020.073  000.051  000.051: sourcing /usr/local/share/nvim/runtime/indent.vim
020.331  000.130  000.130: sourcing /usr/local/share/nvim/runtime/syntax/synload.vim
020.492  000.354  000.224: sourcing /usr/local/share/nvim/runtime/syntax/syntax.vim
024.954  019.034  006.949: sourcing /home/alexander/.config/nvim/init.vim
024.963  000.065: sourcing vimrc file(s)
026.255  000.983  000.983: sourcing /home/alexander/.vim/plugged/nvim-treesitter/plugin/nvim-treesitter.vim
027.048  000.427  000.427: sourcing /usr/local/share/nvim/runtime/plugin/gzip.vim
027.088  000.017  000.017: sourcing /usr/local/share/nvim/runtime/plugin/health.vim
027.288  000.157  000.157: sourcing /usr/local/share/nvim/runtime/plugin/man.vim
027.946  000.268  000.268: sourcing /usr/local/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
028.105  000.796  000.528: sourcing /usr/local/share/nvim/runtime/plugin/matchit.vim
028.334  000.208  000.208: sourcing /usr/local/share/nvim/runtime/plugin/matchparen.vim
028.919  000.564  000.564: sourcing /usr/local/share/nvim/runtime/plugin/netrwPlugin.vim
029.153  000.189  000.189: sourcing /usr/local/share/nvim/runtime/plugin/rplugin.vim
029.319  000.135  000.135: sourcing /usr/local/share/nvim/runtime/plugin/shada.vim
029.397  000.033  000.033: sourcing /usr/local/share/nvim/runtime/plugin/spellfile.vim
029.621  000.194  000.194: sourcing /usr/local/share/nvim/runtime/plugin/tarPlugin.vim
029.764  000.110  000.110: sourcing /usr/local/share/nvim/runtime/plugin/tohtml.vim
029.818  000.026  000.026: sourcing /usr/local/share/nvim/runtime/plugin/tutor.vim
030.088  000.246  000.246: sourcing /usr/local/share/nvim/runtime/plugin/zipPlugin.vim
030.428  001.383: loading rtp plugins
030.628  000.200: loading packages
031.152  000.524: loading after plugins
031.169  000.018: inits 3
036.262  005.093: reading ShaDa
039.895  000.399  000.399: sourcing /usr/local/share/nvim/runtime/autoload/dist/ft.vim
040.166  000.130  000.130: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
040.501  000.011  000.011: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
040.943  000.032  000.032: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
042.554  000.248  000.248: sourcing /usr/local/share/nvim/runtime/syntax/javascript.vim
044.209  001.530  001.530: sourcing /usr/local/share/nvim/runtime/syntax/vb.vim
048.303  004.023  004.023: sourcing /usr/local/share/nvim/runtime/syntax/css.vim
049.151  007.562  001.761: sourcing /usr/local/share/nvim/runtime/syntax/html.vim
049.452  008.361  000.799: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
095.710  000.103  000.103: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
096.052  000.007  000.007: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
096.381  000.020  000.020: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
097.512  000.193  000.193: sourcing /usr/local/share/nvim/runtime/syntax/javascript.vim
098.936  001.332  001.332: sourcing /usr/local/share/nvim/runtime/syntax/vb.vim
103.247  004.255  004.255: sourcing /usr/local/share/nvim/runtime/syntax/css.vim
104.054  007.265  001.485: sourcing /usr/local/share/nvim/runtime/syntax/html.vim
104.327  007.850  000.585: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
105.793  000.199  000.199: sourcing /usr/local/share/nvim/runtime/syntax/javascript.vim
107.170  001.288  001.288: sourcing /usr/local/share/nvim/runtime/syntax/vb.vim
111.538  004.313  004.313: sourcing /usr/local/share/nvim/runtime/syntax/css.vim
112.386  007.308  001.507: sourcing /usr/local/share/nvim/runtime/syntax/html.vim
112.664  007.895  000.587: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
113.222  000.096  000.096: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
113.460  000.006  000.006: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
113.785  000.019  000.019: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
114.906  000.193  000.193: sourcing /usr/local/share/nvim/runtime/syntax/javascript.vim
116.286  001.290  001.290: sourcing /usr/local/share/nvim/runtime/syntax/vb.vim
120.648  004.307  004.307: sourcing /usr/local/share/nvim/runtime/syntax/css.vim
121.528  007.340  001.549: sourcing /usr/local/share/nvim/runtime/syntax/html.vim
121.809  007.930  000.590: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
123.150  000.188  000.188: sourcing /usr/local/share/nvim/runtime/syntax/javascript.vim
124.569  001.331  001.331: sourcing /usr/local/share/nvim/runtime/syntax/vb.vim
128.938  004.296  004.296: sourcing /usr/local/share/nvim/runtime/syntax/css.vim
129.907  007.458  001.643: sourcing /usr/local/share/nvim/runtime/syntax/html.vim
130.186  008.048  000.589: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
130.559  053.391: opening buffers
130.575  000.016: BufEnter autocommands
130.578  000.002: editing files in windows
130.652  000.074: VimEnter autocommands
130.654  000.002: UIEnter autocommands
130.655  000.001: before starting main loop
767.929  637.273: first screen update
767.932  000.004: --- NVIM STARTED ---
```

**0.5.1**
```
times in msec
 clock   self+sourced   self:  sourced script
 clock   elapsed:              other lines

000.008  000.008: --- NVIM STARTING ---
000.267  000.258: locale set
000.539  000.272: inits 1
000.552  000.013: window checked
000.599  000.047: parsing arguments
000.654  000.055: expanding arguments
000.690  000.036: inits 2
001.091  000.401: init highlight
001.092  000.001: waiting for UI
001.926  000.834: done waiting for UI
001.943  000.017: init screen for UI
003.418  001.333  001.333: sourcing /home/alexander/.local/share/nvim/site/autoload/plug.vim
009.602  000.070  000.070: sourcing /home/alexander/.vim/plugged/vim-elixir/ftdetect/elixir.vim
009.824  000.023  000.023: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/elixir.vim
009.847  000.010  000.010: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fennel.vim
009.876  000.015  000.015: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fish.vim
009.900  000.011  000.011: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/fusion.vim
009.926  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gdresource.vim
009.948  000.010  000.010: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gdscript.vim
009.972  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/glimmer.vim
010.010  000.025  000.025: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/glsl.vim
010.036  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/gomod.vim
010.065  000.016  000.016: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/graphql.vim
010.094  000.016  000.016: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/hcl.vim
010.116  000.009  000.009: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/heex.vim
010.138  000.009  000.009: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/hjson.vim
010.164  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/json5.vim
010.191  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/julia.vim
010.220  000.015  000.015: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/ledger.vim
010.244  000.011  000.011: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/nix.vim
010.267  000.011  000.011: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/prisma.vim
010.290  000.010  000.010: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/pug.vim
010.327  000.014  000.014: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/ql.vim
010.362  000.020  000.020: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/query.vim
010.401  000.023  000.023: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/surface.vim
010.425  000.011  000.011: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/teal.vim
010.459  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/tlaplus.vim
010.483  000.012  000.012: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/yang.vim
010.509  000.013  000.013: sourcing /home/alexander/.vim/plugged/nvim-treesitter/ftdetect/zig.vim
011.254  006.865  006.437: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/filetype.vim
011.547  000.038  000.038: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/ftplugin.vim
011.844  000.031  000.031: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/indent.vim
012.437  000.249  000.249: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/syncolor.vim
012.731  000.601  000.353: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/synload.vim
012.811  000.743  000.141: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/syntax.vim
020.306  018.254  009.246: sourcing /home/alexander/.config/nvim/init.vim
020.315  000.118: sourcing vimrc file(s)
021.473  000.907  000.907: sourcing /home/alexander/.vim/plugged/nvim-treesitter/plugin/nvim-treesitter.vim
022.136  000.327  000.327: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/gzip.vim
022.161  000.009  000.009: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/health.vim
022.277  000.104  000.104: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/man.vim
022.721  000.179  000.179: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/pack/dist/opt/matchit/plugin/matchit.vim
022.809  000.516  000.337: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/matchit.vim
022.959  000.137  000.137: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/matchparen.vim
023.323  000.349  000.349: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/netrwPlugin.vim
023.466  000.121  000.121: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/rplugin.vim
023.583  000.097  000.097: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/shada.vim
023.625  000.022  000.022: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/spellfile.vim
023.774  000.129  000.129: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/tarPlugin.vim
023.865  000.071  000.071: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/tohtml.vim
023.901  000.017  000.017: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/tutor.vim
024.075  000.157  000.157: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/plugin/zipPlugin.vim
024.304  001.029: loading plugins
024.433  000.129: loading packages
024.668  000.235: loading after plugins
024.680  000.012: inits 3
028.381  003.702: reading ShaDa
079.298  000.624  000.624: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/autoload/dist/ft.vim
079.760  000.211  000.211: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
080.537  000.019  000.019: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
084.199  000.054  000.054: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
088.125  000.647  000.647: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/javascript.vim
092.250  003.817  003.817: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/vb.vim
100.304  007.849  007.849: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/css.vim
102.088  016.055  003.742: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/html.vim
103.202  018.150  002.094: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
136.711  000.087  000.087: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
137.046  000.006  000.006: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
138.626  000.019  000.019: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
140.487  000.282  000.282: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/javascript.vim
142.501  001.878  001.878: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/vb.vim
146.752  004.152  004.152: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/css.vim
147.554  008.088  001.775: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/html.vim
148.001  009.015  000.927: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
150.498  000.308  000.308: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/javascript.vim
152.350  001.708  001.708: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/vb.vim
156.445  004.013  004.013: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/css.vim
157.345  007.915  001.886: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/html.vim
157.805  008.874  000.959: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
158.881  000.085  000.085: sourcing /home/alexander/.vim/plugged/vim-elixir/ftplugin/elixir.vim
159.211  000.006  000.006: sourcing /home/alexander/.vim/plugged/nvim-treesitter/after/ftplugin/elixir.vim
160.839  000.019  000.019: sourcing /home/alexander/.vim/plugged/vim-elixir/indent/elixir.vim
162.669  000.279  000.279: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/javascript.vim
164.547  001.740  001.740: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/vb.vim
168.562  003.932  003.932: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/css.vim
169.356  007.660  001.709: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/html.vim
169.845  008.642  000.982: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
172.184  000.301  000.301: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/javascript.vim
174.081  001.758  001.758: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/vb.vim
178.132  003.967  003.967: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/css.vim
178.937  007.749  001.723: sourcing /home/alexander/.asdf/installs/neovim/0.5.1/share/nvim/runtime/syntax/html.vim
179.401  008.704  000.955: sourcing /home/alexander/.vim/plugged/vim-elixir/syntax/elixir.vim
180.321  097.426: opening buffers
180.342  000.022: BufEnter autocommands
180.344  000.002: editing files in windows
180.401  000.057: VimEnter autocommands
180.403  000.002: UIEnter autocommands
180.404  000.001: before starting main loop
560.775  380.371: first screen update
560.780  000.005: --- NVIM STARTED ---
```

I'm not sure how to ensure tree-sitter is being compiled into a release build, but I imagine compiling neovim with ```make CMAKE_BUILD_TYPE=RelWithDebInfo``` does so. Is that correct?

**Comment by maxbrunsfeld** [comment]
Ok, I believe this is fixed in Tree-sitter 0.20.2.

**Comment by razzeee** [comment]
Did you mean to only tag and not release? https://github.com/tree-sitter/tree-sitter/releases
